#!/usr/bin/env python
PACKAGE = "people_publisher"

import sys
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("frame_id",    str_t,    0, "frame_id", "map")
gen.add("person_name",    str_t,    0, "person_name", "Oscar")

gen.add("centerShiftX",    double_t,    0, "proxemic: centerShiftX", 0,  -10, 10)
gen.add("centerShiftY",    double_t,    0, "proxemic: centerShiftY", 0,  -10, 10)
gen.add("rotation",    double_t,    0, "proxemic: rotation", 0,  -10, 10)
gen.add("spreadX",    double_t,    0, "proxemic: spreadX", 1,  -10, 10)
gen.add("spreadY",    double_t,    0, "proxemic: spreadX", 1,  -10, 10)

gen.add("velocityX",    double_t,    0, "person: velocity in X", 0,  -10, 10)
gen.add("velocityY",    double_t,    0, "person: velocity in Y", 0,  -10, 10)
gen.add("velocityZ",    double_t,    0, "person: velocity in Z", 0,  -10, 10)

gen.add("positionX",  double_t,    0, "person: position X", 1.5,  -10, 10)
gen.add("positionY",    double_t,    0, "person: position Y", 0,  -10, 10)
gen.add("positionZ",    double_t,    0, "person: position Z", 0,  -10, 10)

gen.add("rotationX",  double_t,    0, "person: rotation X", 0,  -10, 10)
gen.add("rotationY",    double_t,    0, "person: rotation Y", 0,  -10, 10)
gen.add("rotationZ",    double_t,    0, "person: rotation Z", 0,  -10, 10)

gen.add("lethalBorder",    int_t,    0, "max cost that does not count as lethal obstacle", 150,  0, 253)
gen.add("freeBorder",    int_t,    0, "min cost to get in costmaps", 30,  0, 253)

exit(gen.generate(PACKAGE, "people_publisher", "PeoplePublisher"))
